catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeImpl.java

```diff
@@ -119,26 +136,16 @@ public MetacardTypeImpl(String name, MetacardType metacardType,
      * @param metacardTypes the set of {@link MetacardType}s to compose this {@code MetacardTypeImpl}
      */
     public MetacardTypeImpl(String name, List<MetacardType> metacardTypes) {
-        this.name = name;
-        List<String> attributeNames = new ArrayList<>();
-        addAttributeDescriptors(attributeNames, new CoreAttributes().getAttributeDescriptors());
-        addAttributeDescriptors(attributeNames, new SecurityAttributes().getAttributeDescriptors());
+        notEmpty(metacardTypes, "The set of Metacard Types cannot be null or empty");
 
-        if (metacardTypes != null) {
-            metacardTypes.forEach(metacardType -> addAttributeDescriptors(attributeNames,
-                    metacardType.getAttributeDescriptors()));
-        }
-    }
+        this.name = name;
 
-    private List<String> addAttributeDescriptors(List<String> attributeNames,
-            Set<AttributeDescriptor> attributeDescriptors) {
-        attributeDescriptors.forEach(attributeDescriptor -> {
-            if (!attributeNames.contains(attributeDescriptor.getName())) {
-                descriptors.add(attributeDescriptor);
-                attributeNames.add(attributeDescriptor.getName());
-            }
-        });
-        return attributeNames;
+        this.descriptors = Stream.concat(metacardTypes.stream(),
```

Comment:
Did you intentionally swap the order of Core and Security attributes to be at the end?  This change will allow them to redefine/override the core and security attributes.

>Creates a lazily concatenated stream whose elements are all the elements of the first stream followed by all the elements of the second stream.
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#concat-java.util.stream.Stream-java.util.stream.Stream-